File: backend/src/main/java/com/awais/hr/module/compliance/service/ComplianceManagementService.java
package com.awais.hr.module.compliance.service;

import java.util.List;
import java.util.Map;

public interface ComplianceManagementService {
    List<Map<String, Object>> getChecklists();
    Map<String, Object> createChecklist(Map<String, Object> body);
    List<Map<String, Object>> getRiskAssessments();
    Map<String, Object> createRiskAssessment(Map<String, Object> body);
    List<Map<String, Object>> getPolicyAcknowledgements();
    Map<String, Object> acknowledgePolicy(String email, Map<String, Object> body);
}

File: frontend/src/services/complianceManagementService.js
import apiClient from './api';

export const getChecklists = () => apiClient.get('/suite/compliance-management/checklists');
export const createChecklist = (dto) => apiClient.post('/suite/compliance-management/checklists', dto);
export const getRiskAssessments = () => apiClient.get('/suite/compliance-management/risks');
export const createRiskAssessment = (dto) => apiClient.post('/suite/compliance-management/risks', dto);
export const getPolicyAcknowledgements = () => apiClient.get('/suite/compliance-management/acknowledgements');
export const acknowledgePolicy = (dto) => apiClient.post('/suite/compliance-management/acknowledge', dto);
